File: Information Retrieval A1/cpp/Doc_class.cpp
/* 
 * File:   Doc_class.cpp
 * Author: z
 *
 * Created on March 22, 2017, 10:28 PM
 */

#include "Doc_class.h"
//package code;

//import java.util.*;
using namespace std;
/*
    string raw_text = "";
    string text = "";
    string docid = "";
    int id = 0;
    string headline = "";
    bool null = true;
    vector<string> tokens;// = NULL;
    vector<string> cleaned_tokens; //= NULL;
 * */
Doc_class::Doc_class() {
    //null = false;
}

Doc_class::Doc_class(int id)
{
    this->id = id;
    this->null = false;
}

Doc_class::Doc_class(const Doc_class& orig) {
    this->id = orig.id;
    this-> raw_text = orig.raw_text;
    this->text = orig.text;
    this->docid = orig.docid;
    this->headline = orig.headline;
    this->null = orig.null;
    this->tokens = orig.tokens;
    this->cleaned_tokens = orig.cleaned_tokens;
}

Doc_class::~Doc_class() {
}

string Doc_class::toString()
{
    string str = "";
//    str = "id: " + this->id + "\n";
    //str = "raw_text length: " + this->raw_text.length() + "\n";
    //str = "text length: " + this->text.length() + "\n";
    return str;
}

File: Information Retrieval A1/php/doc_class.php
<?php

class doc {
    public $raw_text = "";
    public $text = "";
    public $docid = "";
    public $id = 0;
    public $headline = "";
    public $tokens = array();
    public $cleaned_tokens = array();

    public function __construct($array) {
        foreach ($array as $key => $value) {
            $this->$key = $value;
        }
    }

    public function __destruct() {
        if (isset($this->raw_text)) {
            $this -> raw_text = null;
            unset($this->raw_text);
        }
        if (isset($this->text)) {
            $this -> text = null;
            unset($this->text);
        }

        if (isset($this->tokens)) {

            foreach ($this->tokens as $token_key => $token) {
                $this->tokens[$token_key] = null;
                unset($this->tokens[$token_key]);
            }

            $this->tokens = null;
            unset($this->tokens);
        }

        if (isset($this->cleaned_tokens)) {

            foreach ($this->cleaned_tokens as $cleaned_token_key => $cleaned_token) {
                $this->cleaned_tokens[$cleaned_token_key] = null;
                unset($this->cleaned_tokens[$cleaned_token_key]);
            }

            $this->cleaned_tokens = null;
            unset($this->cleaned_tokens);
        }

        unset($this->headline);

        unset($this->docid);
        
        unset($this);

        //gc_collect_cycles();
    }

}
